### DISCUSSION TITLE
worker_autoscale not respected

### DISCUSSION THREAD

**Discussion by rdjouder** [apache/airflow#29867] [Q&A]
Hello I am using Airflow 2.4.1 with celery executor,
celery_app_name = airflow.executors.celery_executor
I am also using worker_autoscale. From container size or flavor used for MLWorkers, I found out that each such container matches 20 running slots. So I have set auto scale:
worker_autoscale = 20 , 2
Every task is assigned to the default pool
default_pool_task_slot_count = 820
Since I can have up to 41 MLWorkers.
The point is that even on small load, 226 running slots on default pool I can see that MLWorkers may have up to 23 concurrent celery python processes/jobs. Going beyond 20 celery jobs triggers a memory usage alert. Is there an other configuration parameter I can try to enforce this autoscale limit ?
>ps -ef |grep cele |wc -l 
23 

Thanks a lot in advance.

**Comment by potiuk** [comment]
I think you might want to look closer and look at what celery processes do. The celery has at least one main process that monitors running processes, so your `ps -ef |grep cele |wc -l` might yield more than just "working processes". 

You can read more about how celery works when it comes to processes in https://docs.celeryq.dev/en/stable/userguide/workers.html
